packages/broker/src/config/ConfigWizard.ts

```diff
@@ -1,254 +1,539 @@
-import inquirer, { Answers } from 'inquirer'
-import { Wallet } from 'ethers'
-import path from 'path'
-import { writeFileSync, existsSync, mkdirSync, chmodSync } from 'fs'
+import { checkbox, confirm, input, password, select } from '@inquirer/prompts'
+import { config as cfg, config } from '@streamr/config'
+import { toEthereumAddress } from '@streamr/utils'
 import chalk from 'chalk'
-import { v4 as uuid } from 'uuid'
-
-import * as WebsocketConfigSchema from '../plugins/websocket/config.schema.json'
+import { BigNumber, Wallet, providers, utils } from 'ethers'
+import { isAddress } from 'ethers/lib/utils'
+import { chmodSync, existsSync, mkdirSync, writeFileSync } from 'fs'
+import { produce } from 'immer'
+import path from 'path'
+import { z } from 'zod'
+import { generateMnemonicFromAddress } from '../helpers/generateMnemonicFromAddress'
 import * as MqttConfigSchema from '../plugins/mqtt/config.schema.json'
+import * as WebsocketConfigSchema from '../plugins/websocket/config.schema.json'
 import * as BrokerConfigSchema from './config.schema.json'
-import { getDefaultFile } from './config'
-import { CURRENT_CONFIGURATION_VERSION, formSchemaUrl } from '../config/migration'
-import { generateMnemonicFromAddress } from '../helpers/generateMnemonicFromAddress'
-import { toEthereumAddress } from '@streamr/utils'
+import { ConfigFile, getDefaultFile } from './config'
 
-export interface PrivateKeyAnswers extends Answers {
-    generateOrImportPrivateKey: 'Import' | 'Generate'
-    importPrivateKey?: string
-}
+const MinBalance = utils.parseEther('0.1')
+
+export const start = async (): Promise<void> => {
+    log(`
+        >
+        > ***Welcome to the Streamr Network!***
+        > This Config Wizard will help you setup your node.
+        >
+        > The steps are documented here:
+        > *https://docs.streamr.network/guides/how-to-run-streamr-node#config-wizard*
+        >
+    `)
+
+    try {
+        const privateKey = await getPrivateKey()
+
+        const nodeAddress = new Wallet(privateKey).address
+
+        const network = await getNetwork()
+
+        const operator = await getOperatorAddress()
+
+        const operatorPlugins = operator ? {
+            operator: {
+                operatorContractAddress: operator,
+            },
+        } : {}
+
+        const { http, ...pubsubPlugins } = await getPubsubPlugins()
+
+        if (http) {
+            Object.assign(pubsubPlugins, {
+                http: {},
+            })
+        }
```

Comment:
Why is this needed?

Reply:
IIRC and IIUC :P including a plugin with default settings is done by adding an empty object like `http: {}` to the plugins section. I can't recall the exact reasoning behind it all though, perhaps we can look at documenting it together when Eric gets back?

Reply:
The comment is very bad, agree. The summary of Slack discussion:
- for each plugin there should be a block inside plugins block, i.e. if `httpPlugin` enabled, then there is a `http` block
- typically the non-default port is defined inside the plugin's block
  - if a default is used, no port information need to be written as we'll use defaults from the plugin schema, e.g. https://github.com/streamr-dev/network/blob/e00d704440bc68d4455c8a614442c310db9684b4/packages/broker/src/plugins/mqtt/config.schema.json#L11
- the only exception is `httpPlugin`: in that case the port is defined in the `httpServer` root level block
 - if we use default port, then no need to define the root level `httpServer` block (as empty is the default value for it in the Broker schema, and that way it uses the default port automatically) https://github.com/streamr-dev/network/blob/e00d704440bc68d4455c8a614442c310db9684b4/packages/broker/src/config/config.schema.json#L59-L60

Reply:
Maybe we wouldn’t need any exceptional cases at lines `50`-`56`, i.e. we’d have just
```ts
const pubsubPlugins = await getPubsubPlugins()
```
and then when we write pubsubPlugins in plugins, we’d omit the `httpPlugin`’s port:
```ts
plugins: {
    ...operatorPlugins.map((p) => p === ‘http’ ? omit(p, ’port) : p),
    ...pubsubPlugins,
},
```
That way it would a bit clearer that this whole thing is about the port, and if getPubsubPlugins start to give other options in the future, we don’t accidentally omit all httpPlugin options at line 54 (where we currently

Reply:
That `map` is ugly af, lol. Plus, `pubsubPlugins` is an object. I got your point about the other fields we potentially lose here though. I'll get that fixed.